src/SSVOpenHexagon/Core/MenuGame.cpp

```diff
@@ -638,6 +678,56 @@ void MenuGame::initInput()
             }
         },
         t::Once);
+    game.addInput(
+        {{k::F5}},
+        [this](ssvu::FT /*unused*/) { reloadLevelAssets(); },t::Once);
+}
+
+void MenuGame::reloadLevelAssets()
+{
+    if(state != States::SMain || !dialogText.empty() || !Config::getDebug()) return;
+    
+    noActions = 2;
+    assets.playSound("beep.ogg");
+
+    string reloadOutput = assets.reloadLevelData(levelData->packId, levelData->packPath, levelData->id);
+    if(reloadOutput != "invalid level folder path\n" && reloadOutput != "no matching level data file found\n")
+    {
+        setIndex(currentIndex); // loads the new levelData
+
+        reloadOutput += assets.reloadMusicData(levelData->packId, levelData->packPath, levelData->musicId);
+        reloadOutput += assets.reloadStyleData(levelData->packId, levelData->packPath, levelData->styleId);
+
+        if(levelData->musicId != "nullMusicId")
+            reloadOutput += assets.reloadMusic(levelData->packId, levelData->packPath, levelData->musicId);
+
+        if(levelData->soundId != "nullSoundId")
+            reloadOutput += assets.reloadCustomSounds(levelData->packId, levelData->packPath, levelData->soundId);
+    }
+
+    reloadOutput += "\n";
+    reloadOutput += "press any key to close this message\n";
+    Utils::uppercasify(reloadOutput);
+
+    txtDialog.setCharacterSize(26);
+    txtDialog.setString(reloadOutput);
+    dialogWidth = getGlobalWidth(txtDialog);
+    dialogHeight = getGlobalHeight(txtDialog);
+    txtDialog.setString("A");
+    lineHeight = getGlobalHeight(txtDialog);
+    frameOffset = 10.f;
+
+    string temp;
+    for(auto& c : reloadOutput)
```

Comment:
```suggestion
    for(char c : reloadOutput)
```

Using `auto` and a reference here are both unnecessary and just obfuscate the fact that `c` is a `char`.

Reply:
I see, i went with that formatting cause it is used in uppercasify
`inline void uppercasify(std::string& s)
{
    for(auto& c : s)
    {
        c = std::toupper(c);
    }
}`

Reply:
Formatting code blocks isn't that hard. You type three backticks on one line, then the language name, and then at the end of it you type three backticks again. Like this:
```c++
inline void uppercasify(std::string& s)
{
    for(auto& c : s)
    {
        c = std::toupper(c);
    }
}
```

Reply:
Got it